### PR TITLE
Fix some of the CppCheck warnings.

### PR DESCRIPTION
**Pull request by TomFischer**

### CODE REVIEW

File: Applications/DataExplorer/VtkVis/VtkVisPipelineView.cpp [hunk 1]
```diff
@@ -85,26 +85,25 @@ void VtkVisPipelineView::contextMenuEvent( QContextMenuEvent* event )
...
         QAction* addFilterAction = menu.addAction("Add filter...");
 
-        QAction* addLUTAction(nullptr);
-        QAction* addMeshingAction(nullptr);
         if (objectType == VTK_IMAGE_DATA)
         {
             // this exception is needed as image object are only displayed in the vis-pipeline
             isSourceItem = false;
-            addMeshingAction = menu.addAction("Convert Image to Mesh...");
+            QAction* addMeshingAction =
```
Comment (rinkk) [inline Applications/DataExplorer/VtkVis/VtkVisPipelineView.cpp:92]: Have you tested this? I suspect this won't work if the UI file doesn't find the QActions to build the menu?!  
(Some for all the other QAction-ptrs.)

File: Applications/DataExplorer/VtkVis/VtkVisTabWidget.cpp [hunk 2]
```diff
@@ -326,16 +326,15 @@ void VtkVisTabWidget::buildProportiesDialog(VtkVisPipelineItem* item)
...
             if (!values.empty())
             {
                 QList<QString> valuesAsString;
                 foreach (QVariant variant, values)
                 valuesAsString.push_back(variant.toString());
 
-                vectorEdit = new VtkAlgorithmPropertyVectorEdit(valuesAsString, key,
-                                                                values.front().type(),
-                                                                algProps);
+                VtkAlgorithmPropertyVectorEdit* vectorEdit =
```
Comment (rinkk) [inline Applications/DataExplorer/VtkVis/VtkVisTabWidget.cpp:335]: Have you tested this? Does the UI still find the array?

File: Applications/DataExplorer/VtkVis/VtkVisPipelineView.cpp (hunk 1 above)
Comment (TomFischer) [reply Applications/DataExplorer/VtkVis/VtkVisPipelineView.cpp:92]: I compiled the DataExplorer and the testrunner. The testrunner worked fine. Is there another test for this?

File: MeshLib/convertMeshToGeo.cpp [hunk 3]
```diff
@@ -133,12 +133,12 @@ bool convertMeshToGeo(const MeshLib::Mesh& mesh,
         addElementToSurface(*elements[i], id_map, *(*sfcs)[surfaceId]);
     }
 
-    std::for_each(sfcs->begin(), sfcs->end(), [](GeoLib::Surface* sfc) {
+    std::for_each(sfcs->begin(), sfcs->end(), [](GeoLib::Surface*& sfc) {
         if (sfc->getNumberOfTriangles() == 0)
         {
             delete sfc;
+            sfc = nullptr;
```
Comment (rinkk) [inline MeshLib/convertMeshToGeo.cpp:140]: This is tricky. *Technically* it shouldn't be possible to add surfaces without triangles. So most likely the test here is not necessary. However, if this *does* happen, then the surface shouldn't be removed here. It'll mess up the Data Explorer if there's suddenly a nullptr in a loaded geometry out of nowhere.

Comment (TomFischer) [reply MeshLib/convertMeshToGeo.cpp:140]: Yes, that is true. But two lines below the vector is cleaned up. The problem here is: The surface containing zero triangles is deleted, but the pointer isn't changed. So the code below doesn't tidy up the vector. IMHO it is possible to remove the entire `for_each` construct and the following lines. What do you think?

File: Applications/DataExplorer/VtkVis/VtkVisPipelineView.cpp (hunk 1 above)
Comment (bilke) [reply Applications/DataExplorer/VtkVis/VtkVisPipelineView.cpp:92]: I guess this works, as the menu is populated during runtime (`menu.addAction()`).

**Comment by endJunction** [comment]
Nice! :fast_forward:

File: Applications/DataExplorer/VtkVis/VtkVisTabWidget.cpp (hunk 2 above)
Comment (endJunction) [reply Applications/DataExplorer/VtkVis/VtkVisTabWidget.cpp:335]: Is this (and actually the same question above) still open, or can we merge?

Comment (rinkk) [reply Applications/DataExplorer/VtkVis/VtkVisTabWidget.cpp:335]: @endJunction I haven't had any feedback regarding the PropertyVector.

Tom tested the QAction-thing and it seems to be working fine.

File: MeshLib/MeshGenerators/MeshGenerator.cpp [hunk 4]
```diff
@@ -30,19 +30,30 @@ std::vector<MeshLib::Node*> MeshGenerator::generateRegularNodes(
     const std::vector<const std::vector<double>*> &vec_xyz_coords,
     const MathLib::Point3d& origin)
 {
+    auto const shift_coordinates = [](auto const& in, auto& out,
```
[the remark this answers is not in the document]
Comment (endJunction) [reply MeshLib/MeshGenerators/MeshGenerator.cpp:33]: all elements of `in` are shifted by `shift` distance and the result is stored in `out`.
On it self it does nothing - it's just a variable holding a lambda. It is called below (line 42) to transform coordinates.

File: Applications/DataExplorer/VtkVis/VtkVisTabWidget.cpp (hunk 2 above)
Comment (TomFischer) [reply Applications/DataExplorer/VtkVis/VtkVisTabWidget.cpp:335]: @rinkk What are the necessary steps to test if the changes are okay?

Comment (rinkk) [reply Applications/DataExplorer/VtkVis/VtkVisTabWidget.cpp:335]: Apply any filter with a multi-value parameter to a dataset to see if the widgets are displayed correctly.

E.g. apply a threshold filter to a mesh and check if two line edits are displayed for the "Range"-parameter.

Comment (TomFischer) [reply Applications/DataExplorer/VtkVis/VtkVisTabWidget.cpp:335]: Tested this, threshold filter works. So my changes don't destroy anything in principle.

**Pull request merged**